File: 02_2017/cpp/Square.cpp
#include "Square.hpp"

#include <cmath>

static const double pi = std::acos(-1);

Square::Square(double sideLength) : Tetragon(sideLength, sideLength, sideLength, sideLength, pi / 2) {}

double Square::surface_area() const noexcept { return sideLengths[0] * sideLengths[0]; }

double Square::perimeter() const noexcept { return 4 * sideLengths[0]; }

File: 03/Shapes/src/main/java/com/dabek/jakub/shapes/Square.java
package com.dabek.jakub.shapes;

public class Square extends Tetragon {

    public Square(double sideLength) {
        super(sideLength, sideLength, sideLength, sideLength, Math.PI / 2);
    }

    @Override
    public double surfaceArea() {
        return sideLengths[0] * sideLengths[0];
    }

    @Override
    public double perimeter() {
        return 4 * sideLengths[0];
    }

}
